python/graphstorm/eval/evaluator.py

```diff
@@ -1023,8 +1023,8 @@ def compute_score(
             if metric == "amri":
                 assert candidate_sizes, \
                     f"candidate_sizes needs to have a value for AMRI, got {candidate_sizes=}."
-                for _, size in candidate_sizes.items():
-                    sizes_list.append(size)
+                for etype, _ in rankings.items():
+                    sizes_list.append(candidate_sizes[etype])
```

Comment:
I think we need to add a TODO or a HINT here: if in the future, there is another metric that needs sizes_list, we should avoid adding the size of each candidate twice.